Utilities/Thread.cpp

```diff
@@ -1774,13 +1776,15 @@ void thread_base::initialize(bool(*wait_cb)(const void*))
 #endif
 
 #if defined(__APPLE__)
-	pthread_setname_np(m_name.get().substr(0, 15).c_str());
+	name.resize(std::min<std::size_t>(15, name.size()));
+	pthread_setname_np(name.c_str());
 #elif defined(__DragonFly__) || defined(__FreeBSD__) || defined(__OpenBSD__)
-	pthread_set_name_np(pthread_self(), m_name.get().c_str());
+	pthread_set_name_np(pthread_self(), name.c_str());
 #elif defined(__NetBSD__)
-	pthread_setname_np(pthread_self(), "%s", const_cast<char*>(m_name.get().c_str()));
+	pthread_setname_np(pthread_self(), "%s", const_cast<char*>(name.c_str()));
```

Comment:
```suggestion
	pthread_setname_np(pthread_self(), "%s", name.data());
```
Because `name` is longer a const string.